### PR TITLE
Add detach-device-alias test for hostdev,controller,redirdev and channel

### PR DESCRIPTION
**Pull request by kylazhang**
Signed-off-by: Kylazhang <weizhan@redhat.com>

### CODE REVIEW

**Comment by kylazhang** [comment]
depends on #1722 #1723 #1724

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            return result.stdout_text.rstrip(':')
+        else:
+            test.error("Can not get usb hub info for testing")
+
+    # backup xml
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    backup_xml = vmxml.copy()
+
+    # wait for vm start successfully
+    vm.wait_for_login()
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: We need ensure the guest is running before login in

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+def run(test, params, env):
+    """
+    Test detach-device-alias command with
+    --config, --live, --current
+
+    1. Test hostdev device detach
+    2. Test scsi controller device detach
+    3. Test redirect device detach
+    4. Test channel devices detach
+    5. Test mem detach
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: It seems this line could be removed.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_detach_device_alias.cfg [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        - controller:
+            detach_controller_type = "scsi"
+            detach_controller_mode = "virtio-scsi"
+            detach_check_xml = "<controller type='scsi' index='%s' model='virtio-scsi'>"
+        - redirdev:
+            detach_redirdev_type = "spicevmc"
+            detach_redirdev_bus = "usb"
+            detach_check_xml = "<redirdev"
+        - channel:
+            detach_channel_type = "spicevmc"
```
Comment (dzhengfy) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_detach_device_alias.cfg]: spicevmc channel is not supported on ppc and s390x. Please help exclude them.

**Pull request merged**